internal/xds/rbac/rbac_engine_test.go

```diff
@@ -1089,3 +1701,109 @@ func (sts *ServerTransportStreamWithMethod) SendHeader(md metadata.MD) error {
 func (sts *ServerTransportStreamWithMethod) SetTrailer(md metadata.MD) error {
 	return nil
 }
+
+// A list of audit events that we can configure a logger to push to such that we
+// can access the contents during a test
+var auditEvents []*audit.Event
+
+// An audit logger that will log to the auditEvents slice
+type TestAuditLoggerBuffer struct {
+}
+
+func (logger TestAuditLoggerBuffer) Log(e *audit.Event) {
+	auditEvents = append(auditEvents, e)
+}
+
+// Builds TestAuditLoggerBuffer
+type TestAuditLoggerBufferBuilder struct {
+}
+
+// The required config for TestAuditLoggerBuffer
+type TestAuditLoggerBufferConfig struct {
+	audit.LoggerConfig
+}
+
+func (builder TestAuditLoggerBufferBuilder) ParseLoggerConfig(configJSON json.RawMessage) (config audit.LoggerConfig, err error) {
+	return TestAuditLoggerBufferConfig{}, nil
+}
+
+func (builder TestAuditLoggerBufferBuilder) Build(config audit.LoggerConfig) audit.Logger {
+	return TestAuditLoggerBuffer{}
+}
+
+func (builder TestAuditLoggerBufferBuilder) Name() string {
+	return "TestAuditLoggerBuffer"
+}
+
+// An audit logger to test using a custom config
+type TestAuditLoggerCustomConfig struct {
```

Comment:
These two types of loggers can be combined, right?

Reply:
Are you suggesting `TestAuditLoggerCustomConfig` could be removed entirely, and we could have a different builder for `TestAuditLoggerBuffer` that uses the custom config?